howtos/README.md

```diff
@@ -0,0 +1,71 @@
+# FLAX HOWTOs
+
+## Introduction
+
+FLAX HOWTOs explain how to implement standard techiques in FLAX. For instance,
+the HOWTO for ensembling learning demonstrates what changes should be made to
+the standard MNIST example in order to train an ensemble of models. As such, 
+HOWTOs are simply "diffs".
+
+## List of HOWTOs
+
+In essence, a HOWTO is a set of changes that are applied to the FLAX master 
+branch on Github. HOWTOs are currently simply diff views between HOWTO branches 
+and the master branch. Soon, they will be merged with the rest of our 
+documentation through readthedocs.
+
+Currently the following HOWTOs are available:
+
+* [Ensembling](https://github.com/marcvanzee/flax/compare/prerelease..howto-ensembling?diff=split)
+
+> :warning: We should add the HOWTOs here automatically, or refer to 
+            readthedocs.
+
+## How HOWTOs work
+
+While HOWTOs are shown as changes on a branch, these branches are read-only for
+the users. HOWTOs are represented as diff files in each client as well. For 
+instance, the "ensembling howto" is represented in two ways:
+
+* In the branch `howto-ensembling` (only on Github).
+* In the file `howtos/diffs/howto-ensembling.diff` (on the master branch).
+
+The branch `howto-ensembling` is simply the result of applying 
+`howto/howto-ensembling.diff` to the master branch.
+
+This setup allows us to ensure all local modifications to HOWTOs can be packed
+in diff files locally, pushed to origin and applied to the master branch 
+automatically.
+
+This process is automatic, which means that users usually won't notice this, 
+except if they want to add a new HOWTO, or their changes require making changes
+to HOWTOs. We explain these two cases below.
+
+## Adding a new HOWTO
+
+In order to create a new HOWTO, you can simply make the desired changes local
+changes and run:
+
+```
+./howtos/script/pack_howto_diff.sh
+```
+
+This will pack the changes made on `howto-<name>` into a diff file and stage it
+for committing. It will also revert all changes you made, but you can directly
+recover these changes simply by applying the diff:
+
+```
+git apply <diff_file>
+```
+
+After pushing your commit, the newly submitted diff file will be unpacked on
+Github with a Github action and a new howto branch will be created for it.
+
+The workflow script for the Github action can be found at 
+`.github/workflows/apply-howto-branches.yml`
+
+## Resolving a conflict with an existing HOWTO
+
+> :warning: TODO: This is still in progress. Currently conflicts are resolved by
+            checking whether the Github action fails, and if so, manually making
+            the required fixes.
```

Comment:
Also please add a section (even if it's not written yet -- with a TODO) for "modifying an existing HOWTO"